### DISCUSSION TITLE
Document for OSM mappers what to pay attention to

### DISCUSSION THREAD

**Discussion by RicoElectrico** [organicmaps/organicmaps#6120] [Ideas]
The generator has a quite a long list of hacks to gracefully consume inconsistent OSM data.
There are probably a few instances where OSM mappers could make Organic Maps work better, as not everything can be accounted for automatically.
Maybe we could make a guide (with overpass queries when applicable)?
For example:
- proper access tagging that isn't just car-centric (causes suboptimal routing for pedestrians and bicycles)
- as above, but failing to block off internal service roads
- don't overlay `highway=*` + `area=yes` on top of `highway=*` (use `area:highway=*`)
- if you tag sidewalks as separate lines, do it properly (for example, maintain connectivity)
- use `alt_name=*`, `short_name=*` tags to deliver better search results

Any other such tips?

Obviously, "don't tag for the renderer", but most of such cases are just incomplete mapping and fixing them would be in line with de-facto OSM practice.

**Comment by biodranik** [comment]
Isn't it about highlighting already existing guidelines for OSM mappers in their wiki? You can just mention there _why_ is it important to follow.

**Reply by RicoElectrico** [reply]
Well, yes and no. The audience would be, for example, people who find out about OM on Hacker News and find some erroneous routing.
I do not expect anyone to read OSM Wiki back to back, if it's even possible 😅
Just condensing the common issues that we see reported in this repo that trace back to OSM data, not OM itself.

**Reply by biodranik** [reply]
Actually, it's a great idea to make a detailed list of the most common OSM issues and make a FAQ for our OM users on how they can fix them!
